client/src/main/java/io/pravega/client/netty/impl/RawClient.java

```diff
@@ -42,6 +45,9 @@
     private final Map<Long, CompletableFuture<Reply>> requests = new HashMap<>();
     private final ResponseProcessor responseProcessor = new ResponseProcessor();
     private final AtomicBoolean closed = new AtomicBoolean(false);
+    @Getter
+    private final AtomicLong requesterId = new AtomicLong(IdGenerator.getRequesterId());
```

Comment:
Why do you have a getter on the `AtomicLong`? It is a very bad practice to expose internals to the outside. Someone can do `getRequesterId().set(<new-value-here>)` and mess up with your implementation.

Please write an explicit getter that only returns the contents. Similarly, if you need to update it, write an explicit setter for it.

Reply:
I see you use `updateAndGet` somewhere below. Then please expose the appropriate API on this class that modifies its internal state. Again, you should not expose your internal state and allow others to modify it.

Another reason this is not good is because both in this class and in your outside callers you are using `IdGenerator.getRequesterId`, therefore both of them need to know to use that. if you create an API such as `nextRequestId()` (with no args), that can encapsulate all this logic and will make your code look cleaner.

Reply:
updated the api based on feedback.